src/app/static/src/tests/components/runWorkflow/runWorkflowCreate.test.ts

```diff
@@ -28,18 +93,89 @@ describe(`runWorkflowCreate`, () => {
         expect(wrapper.emitted("create").length).toBe(1)
     })
 
-    it(`can emit run navigation step`, async() => {
+    it(`can emit re-run navigation step`, async (done) => {
         const wrapper = getWrapper()
-        expect(wrapper.find("h2").text()).toBe("Run workflow")
-        await wrapper.find("#rerun").trigger("click")
-        expect(wrapper.emitted("rerun").length).toBe(1)
+
+        setTimeout(async () => {
+            expect(wrapper.find("h2").text()).toBe("Run workflow")
+            expect(mockAxios.history.get.length).toBe(1)
+            expect(mockAxios.history.get[0].url).toBe("http://app/workflows")
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
+            expect(vueSelect.find("input").attributes("placeholder")).toBe("Search by name or user...")
+            await wrapper.setData({runWorkflowMetadata: workflowMetadata[0]})
+            await wrapper.find("#rerun").trigger("click")
+            expect(wrapper.emitted("rerun").length).toBe(1)
+            expect(wrapper.emitted().rerun[0]).toEqual(runnableWorkflowMetadata)
+            done()
+        })
     })
 
-    it(`can emit clone navigation step`, async() => {
+    it(`can emit clone navigation step`, async (done) => {
         const wrapper = getWrapper()
-        expect(wrapper.find("h2").text()).toBe("Run workflow")
-        await wrapper.find("#clone").trigger("click")
-        expect(wrapper.emitted("clone").length).toBe(1)
+
+        setTimeout(async () => {
+            expect(wrapper.find("h2").text()).toBe("Run workflow")
+            expect(mockAxios.history.get.length).toBe(1)
+            expect(mockAxios.history.get[0].url).toBe("http://app/workflows")
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
```

Comment:
Not querying this, but would like to understand: what does this `$emit` do?

Reply:
this triggers selected option to get updated inside vue-select component with the given option...

Reply:
Ah, thanks! Sorry - was confused by the ordering. So in reality the `setData` comes before the `$emit` and is followed by the `trigger`. All good.

Reply:
No issues with ordering, $emit is for the dropdown option while setData below is for another trigger/$emit ..............

Reply:
I get it now - thanks.

It doesn't invalidate this test, but the way it's written does trigger the "An error occurred while retrieving..." message. If you set up the mock Axios to return the workflowMetadata for the selected workflow then it wouldn't, and I guess you wouldn't need to use `setData` manually.